### PR TITLE
Add simple Go server as an app to promote

### PR DESCRIPTION
**Pull request by a-roberts**
Intended to be as lightweight as possible. I noticed performance was worse when this was a multilayer Dockerfile, but this is faster than the previous Node.js app we were using with appmetrics (https://github.com/akihikokuroda/promote-demo).

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,23 @@
...
+
+```
+services promote --from promote-demo --to https://github.com/<myorg>/<mydevopsrepository>.git --service promote-demo
+```
+
+an examplar *GitOps* repository can be found at https://github.com/rhd-gitops-example/gitops-example-staging.
+
+So what happens here, with the promote?
+
+https://docs.google.com/document/d/102LiAQBc0Mb3ZNSguL1ZhCojvEXOGGtzUdub-e58Mog/edit# outlines the design in more detail, but specifically (for the local "source to GitOps" repository), this happens:
```
Comment (mnuttall) [inline README.md]: Our README in the [Overview](https://github.com/rhd-gitops-example/services/blob/master/README.md#services--) links to the now-external version of this doc, [here](https://github.com/rhd-gitops-example/docs/tree/master/model#gitops-repository). Please can you update to link to one of these two?

File: README.md [hunk 2]
```diff
@@ -1,2 +1,23 @@
...
+
+A pull request will be created against the GitOps repository, whereby you will see the `config/deploy.yaml` contents are now in the GitOps repository's `/environments/*environment name*/services/*service name*/base/config` folder. 
+
+In the GitOps repository, is assumed there'll be only one folder under `environments` for now, otherwise you'll have to provide a `--env` flag which I've yet to fully implement and document. If the GitOps repository has `environments/stage`, that would be used as the target folder.
```
Comment (mnuttall) [inline README.md]: I've raised https://github.com/rhd-gitops-example/services/issues/92 to cover this work - you could refer to that?

File: README.md [hunk 3]
```diff
@@ -1,2 +1,23 @@
...
+
+If there are several, you can add `--env prod` for example, and then `environments/prod` will be used as the target folder instead. It is important to note this target folder only exists on the GitOps repository, not your source application's repository.
```
Comment (mnuttall) [inline README.md]: Again I don't think we've finalised a syntax for `--env` so it's maybe best not to refer to it here, beyond a link to issue 92.

**Review by mnuttall — CHANGES_REQUESTED** [review]

File: README.md (hunk 3 above)
Comment (a-roberts) [reply README.md]: Yeah, remnants of the readme from when I did plan to implement this but we noticed the interesting nuances (source vs dev)

File: README.md (hunk 1 above)
Comment (a-roberts) [reply README.md]: Done

**Review by mnuttall — APPROVED** [review]
LGTM thanks @a-roberts

**Pull request merged**